django_api_decorator/types.py

```diff
@@ -35,5 +40,5 @@ def __init__(
     ) -> None:
         super().__init__(*args, *kwargs)
 
-        self.message = errors if errors is not None else [message]
```

Comment:
At the end here you've changed from `[message]` to `message`. Ie. it's no longer an array, but rather just a string. Because of that you've also changed the structure of the responses. See where it's used, we return `.message` directly to the client as `"error"`: 

https://github.com/kolonialno/django-api-decorator/blob/07bdd48ef1f5879ced50f59e536cb07114571df2/django_api_decorator/decorators.py#L160-L163

Reply:
ah, now I follow. And I guess that also explains why we set `self.message` to `errors`

Reply:
+1 to what Sigurd says. Either you pass a custom message, or you give it a form-error-dict or similar. We never use the class with *both* errors and a message.
Message is in simple cases, and errors is when you want a mapping from error per field/section/whatever key in the frontend for customization.
The best would be to have mypy check that you didn't supply both args, but I guess that might be difficult 😅

Reply:
I think we could refactor this a bit to make it easier to use. For example the exception class could implement an `to_response()` method or something like that, and then we can have different exception classes (or you can subclass it if you want custom behavior).

Reply:
Ye, sounds like my use-case is probably a bit special and this is meant for something else. [Maybe I should make my own Error class](https://xkcd.com/927/) /s

Thanks for the clarification, Ill close this